07_errors/willshen8/error.go

```diff
@@ -0,0 +1,13 @@
+package main
+
+import "fmt"
+
+type Error struct {
+	Message string
+	Code    int
+}
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("Error code %d: %s",
+		e.Code, e.Message)
+}
```

Comment:
I like the idea in the example where the error codes are defined as an enum within error.go and then the more meaningful name (eg. ErrNotFound) is passed into the error function rather than just passing a 1 or 2.

Reply:
I was thinking about this one and thought what they asking is an error code similar to network response code like 404, 200, etc. 

The one in the lecture slides `var ErrPermission = errors.New("permission denied")` actually returns a string. And my idea of code is a number, but I could be wrong.

Reply:
yeah, that's good, numeric code is in the example too.